main.tf

```diff
@@ -54,6 +54,7 @@ EOT
 
 # EKS Provider
 provider "kubernetes" {
+  # This defers provider initialization until the cluster is ready
```

Comment:
This seems to have been added for clarity, but the code has been here for some time and the sentence is not really clarifying.

Reply:
It was just intended to point out that the next data reference for the endpoint value, on line 58, will delay the initialization of the k8s provider until that data element has a value and in that way is similar to calling out a depends_on for the entire EKS cluster being ready. Definitely fine with removing it unless adding that makes it more useful.

Reply:
I'll this up to you.

Reply:
Added text to clarify the comment